### ISSUE TITLE
Node update via homebrew resets global packages

### ISSUE THREAD

**Issue by Clashsoft** [pnpm/pnpm#2666]
### Code to reproduce the issue:

```
$ brew install node@14.4.0
$ npm i pnpm
$ pnpm i -g @angular/cli # or anything else, really
$ ng version
10.0.1
$ brew install node@14.5.0 # or brew upgrade
$ ng
zsh: ng: command not found
```

### Expected behavior:

Globally installed packages / global `package.json` are not lost.

### Actual behavior:

The directory `/usr/local/Cellar/node/<old-node-version>` is deleted. A new directory `/usr/local/Cellar/node/<new-node-version` is created, but `/usr/local/Cellar/node/<old-node-version>/pnpm-global/`, which contains the global `package.json`, is lost.

### Workaround

Reinstall all packages previously installed with `pnpm i -g`.

### Additional information:

 - `node -v` prints: 14.5.0 (was 14.4.0)
 - Windows, OS X, or Linux?: macOS

**Comment by zkochan** [comment]
I think this is expected. It would work the same with npm or yarn.

**Comment by Clashsoft** [comment]
npm stores the files of global packages in `/usr/local/lib/node_modules`, which is persistent. I'm not sure where it stores *which* global packages are installed (there seems to be no global `package.json`).

**Comment by zkochan** [comment]
And the commands are created in `/usr/local/bin`?

**Comment by Clashsoft** [comment]
Yes, actually the `ng` command WAS found (because the symlink was not deleted), but it printed errors because it could not find things in `/usr/local/Cellar/node/<old-node-version>/pnpm-global/4/node_modules`.

**Comment by zkochan** [comment]
I guess, we could reuse the location of the global bin directory to find the right location for the global packages...

I think this should be the algorithm. If the global bin directory ends with `bin`, then the packages should be installed to one directory up. For instance, if global bin dir is `/usr/local/bin`, then the global dir is `/usr/local/pnpm-global`.

If the global bin dir does not end with `bin`, then create the global dir inside it. So if the global bin dir is `C:\Program File\NodeJS`, then the global packages will be in `C:\Program FIles\NodeJS\pnpm-global`.

**Comment by Clashsoft** [comment]
What about the [`prefix`](https://docs.npmjs.com/misc/config#global) or `pnpm-prefix` config values?

**Comment by zkochan** [comment]
We do read these config settings of npm but it appears that it doesn't always work. That is why I had to come up with some new solution.

**Comment by Clashsoft** [comment]
Ah I see. I suppose your proposed algorithm should work, then.

**Comment by RajceP** [comment]
Any update on this issue?

**Comment by wildwestrom** [comment]
Just wanted to say, I manually change the pnpm launcher in `~/.local/share/npm-global/bin/pnpm`.
```diff
#!/bin/sh
basedir=$(dirname "$(echo "$0" | sed -e 's,¥¥,/,g')")

case `uname` in
    *CYGWIN*) basedir=`cygpath -w "$basedir"`;;
esac

- "/Users/main/homebrew/Cellar/node/17.0.1/bin/node"  "$basedir/../pnpm-global/5/node_modules/pnpm/bin/pnpm.cjs" 
+ "/Users/main/homebrew/bin/node"  "$basedir/../pnpm-global/5/node_modules/pnpm/bin/pnpm.cjs"
"$@"
exit $?
```

Not sure what causes this to reset, but it's annoying as hell. Currently on node `17.3.0`.

**Comment by gluxon** [comment]
I ended up setting the [`global-dir`](https://pnpm.io/npmrc#global-dir) to workaround this.

```sh
sudo mkdir -p /usr/local/pnpm-global
sudo chown $(whoami) /usr/local/pnpm-global
```
```sh
pnpm config set global-dir /usr/local/pnpm-global
```

**Comment by gluxon** [comment]
Should be fixed https://github.com/pnpm/pnpm/pull/4280. 🙂
